recipes/libx264/all/conanfile.py

```diff
@@ -96,6 +96,7 @@ def _configure_autotools(self):
             # bitstream-a.S:29:18: error: unknown token in expression
             args.append("--extra-asflags=-arch arm64")
             args.append("--extra-ldflags=-arch arm64")
+            args.append("--host=aarch64-apple-darwin")
```

Comment:
Weird that is fix something. It is supposed to be automatically injected by conan.

Reply:
The issue seems to be in `libx264`'s `./configure` script. Passing --host overrides the `config.guess` which is returning "arm-apple-darwin21.1.0". Once all the host parsing happens in `configure` it falls into the `arm` branch which defaults to armv7 instead of the aarch64 branch.

Reply:
This is all in `libx264`'s `./configure`.

```
 550 if [ "x$host" = x ]; then
 551     host="$(${SRCPATH}/config.guess)"
 552 fi
 553 # normalize a triplet into a quadruplet
 554 host="$(${SRCPATH}/config.sub $host)"
 555
 556 # split $host
 557 host_cpu="${host%%-*}"
 558 host="${host#*-}"
 559 host_vendor="${host%%-*}"
 560 host_os="${host#*-}"
```
`$host_cpu` winds up with the value `arm`. Then down in the switch statement at line 706 it falls into the `arm` branch instead of `aarch64`:
```
 791     arm*)
 792         ARCH="ARM"
 793         if [ "$SYS" = MACOSX ] ; then
 794             AS="${AS-${CC}}"
 795             ASFLAGS="$ASFLAGS -DPREFIX -DPIC"  # apple's ld doesn't support movw/movt relocations at all
 796             # build for armv7 by default
 797             if ! echo $CFLAGS | grep -Eq '\-arch' ; then
 798                 CFLAGS="$CFLAGS -arch armv7"
 799                 LDFLAGS="$LDFLAGS -arch armv7"
 800             fi
 801         elif [ "$SYS" = WINDOWS ] && [ "$compiler" = CL ] ; then
 802             AS="${AS-${SRCPATH}/tools/gas-preprocessor.pl -arch arm -as-type armasm -force-thumb -- armasm -nologo -ignore 4509}"
 803         elif [ "$SYS" = WINDOWS ] ; then
 804             AS="${AS-${SRCPATH}/tools/gas-preprocessor.pl -arch arm -as-type clang -force-thumb -- ${CC} -mimplicit-it=always}"
 805         else
 806             AS="${AS-${CC}}"
 807         fi
 808         ;;
 809     aarch64)
 810         ARCH="AARCH64"
 811         stack_alignment=16
 812         if [ "$SYS" = MACOSX ] ; then
 813             AS="${AS-${CC}}"
 814             ASFLAGS="$ASFLAGS -DPREFIX -DPIC"
 815         elif [ "$SYS" = WINDOWS ] && [ "$compiler" = CL ] ; then
 816             AS="${AS-${SRCPATH}/tools/gas-preprocessor.pl -arch aarch64 -as-type armasm -- armasm64 -nologo}"
 817         else
 818             AS="${AS-${CC}}"
 819         fi
 820         ;;
```

Reply:
I see. Maybe a more robust approach would be to override `config.guess` and `config.sub` files with the ones from `gnu-config` recipe? Take a look at `coin-clp` recipe for example.

Reply:
> I see. Maybe a more robust approach would be to override `config.guess` and `config.sub` files with the ones from `gnu-config` recipe? Take a look at `coin-clp` recipe for example.

Perhaps but that would be a change in the `libx264` and outside of conan. Maybe that's the right thing to do. I'm a first-time committer to CCI so I'm not sure what the communities approach is when a build fails in the upstream source. The patch I created seems to work around this build failure given only a conan change (within the remit of this repo) and unblocks a number of dependent builds (I believe opencv depends on libx264) for Apple Silicon. My change is a bit of a hack (perhaps) but it seems benign since we already check that the OS is Macos and the CPU is aarch64.